platform/android/MapboxGLAndroidSDKTestApp/src/androidTest/java/com/mapbox/mapboxsdk/location/LocationComponentTest.kt

```diff
@@ -121,22 +133,30 @@ class LocationComponentTest : EspressoTest() {
     val componentAction = object : LocationComponentAction.OnPerformLocationComponentAction {
       override fun onLocationComponentAction(component: LocationComponent, mapboxMap: MapboxMap,
                                              style: Style, uiController: UiController, context: Context) {
-        component.activateLocationComponent(
-          context,
-          style,
-          null,
-          LocationComponentOptions.builder(context)
-            .staleStateTimeout(200)
-            .enableStaleState(false)
-            .accuracyAlpha(.5f)
-            .accuracyColor(Color.BLUE)
-            .build())
+
+          locationComponentActivationOptions = LocationComponentActivationOptions
+                  .builder(context, style)
+                  .locationEngine(null)
+                  .locationComponentOptions(
+                          LocationComponentOptions.builder(context)
+                                  .staleStateTimeout(200)
+                                  .enableStaleState(false)
+                                  .accuracyAlpha(.5f)
+                                  .accuracyColor(Color.BLUE)
+                                  .build()
+                  )
+                  .build()
+        component.activateLocationComponent(locationComponentActivationOptions)
+
         component.isLocationComponentEnabled = true
 
         val locationEngine = component.locationEngine
         val componentOptions = component.locationComponentOptions
 
-        assertThat(locationEngine, nullValue())
+        // notNullValue() used for LocationEngine because
+        // com.mapbox.android.core.location.LocationEngineProxy
+        // should be returned, rather than `null`
+        assertThat(locationEngine, notNullValue())
```

Comment:
This test should set both `locationEngine` to `null` and `useDefaultEngine` to `false` and still test for the `null` location engine.

Reply:
Ok, now the test is 

```kotlin
 @Test
  fun locationComponent_doesntInitializeEngineWhenNullProvided() {
    validateTestSetup()
    val componentAction = object : LocationComponentAction.OnPerformLocationComponentAction {
      override fun onLocationComponentAction(component: LocationComponent, mapboxMap: MapboxMap,
                                             style: Style, uiController: UiController, context: Context) {

          locationComponentActivationOptions = LocationComponentActivationOptions
                  .builder(context, style)
                  .locationEngine(null)
                  .useDefaultLocationEngine(false)
                  .locationComponentOptions(
                          LocationComponentOptions.builder(context)
                                  .staleStateTimeout(200)
                                  .enableStaleState(false)
                                  .accuracyAlpha(.5f)
                                  .accuracyColor(Color.BLUE)
                                  .build()
                  )
                  .build()
        component.activateLocationComponent(locationComponentActivationOptions)

        component.isLocationComponentEnabled = true

        val locationEngine = component.locationEngine
        val componentOptions = component.locationComponentOptions

        assertThat(locationEngine, nullValue())
        assertThat(componentOptions, notNullValue())

        TestingAsyncUtils.waitForLayer(uiController, idlingResource.mapView)
        assertThat(componentOptions?.accuracyAlpha(), `is`(.5f))
        assertThat(componentOptions?.accuracyColor(), `is`(Color.BLUE))
      }
    }

    executeComponentTest(componentAction)
  }
```